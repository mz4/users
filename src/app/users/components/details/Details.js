import classnames from 'classnames';
import {
  Bar,
  BarChart,
  PolarAngleAxis,
  PolarGrid,
  PolarRadiusAxis,
  Radar,
  RadarChart,
  ResponsiveContainer
} from 'recharts';
import { dataA, dataB } from '../../../data';
import { Avatar, Subtitle, Text, Title, Item } from '../../../library/index';
import styles from './Details.module.scss';

const Details = ({ user }) => {
  const { first_name, avatar } = user;
  return (
    <div className={styles.bodyContainer}>
      <Title text="User Details" customClassName={styles.title} />
      <div className={classnames(styles.box1, styles.avatar)}>
        <Avatar alt={first_name} src={avatar} />
      </div>
      <div className={classnames(styles.box2, styles.details)}>
        <div className={styles.detailsContainer}>
          <Item>
            <Subtitle text="Name" />
            <Text text={first_name} />
          </Item>
          <Item>
            <Subtitle text="Request No." />
            <Text text="2000839625936" />
          </Item>
          <Item>
            <Subtitle text="Connection Name" />
            <Text text="Angela Valdez" />
          </Item>
          <Item>
            <Subtitle text="Date Request" />
            <Text text="23/10/2021" />
          </Item>
          <div className={styles.description}>
            <Item>
              <Subtitle text="Description" />
              <Text
                text="Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                  Integer ac ante varius, vestibulum urna id, tincidunt enim.
                  Vestibulum maximus risus dolor, dapibus finibus velit euismod
                  eget."
              />
            </Item>
          </div>
        </div>
      </div>
      <div className={classnames(styles.box, styles.box3, styles.orders)}>
        <Subtitle text="Total Orders Today" />
        <div className={styles.orderValue}>12.450 $</div>
      </div>
      <div className={classnames(styles.box, styles.box3, styles.orders)}>
        <Subtitle text="Total Clients Today" />
        <div className={styles.orderValue}>25</div>
      </div>
      <div className={classnames(styles.box, styles.box3, styles.orders)}>
        <Subtitle text="Revenue Day Ratio" />
        <div className={styles.orderValue}>25%</div>
      </div>
      <div className={classnames(styles.box4, styles.chart)}>
        <Subtitle text="Chart Area" />
        <ResponsiveContainer width="100%" height="100%">
          <BarChart width={150} height={40} data={dataA}>
            <Bar dataKey="uv" fill="#8884d8" />
          </BarChart>
        </ResponsiveContainer>
      </div>
      <div className={classnames(styles.box4, styles.chart)}>
        <Subtitle text="Chart Billing" />
        <ResponsiveContainer width="100%" height="100%">
          <RadarChart cx="50%" cy="50%" outerRadius="80%" data={dataB}>
            <PolarGrid />
            <PolarAngleAxis dataKey="subject" />
            <PolarRadiusAxis />
            <Radar
              name="Mike"
              dataKey="A"
              stroke="#8884d8"
              fill="#8884d8"
              fillOpacity={0.6}
            />
          </RadarChart>
        </ResponsiveContainer>
      </div>
    </div>
  );
};

export default Details;
