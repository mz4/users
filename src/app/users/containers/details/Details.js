import { Avatar, Title } from '../../../library/index';
import Charts from '../../components/details/Charts';
import styles from './Details.module.scss';
import Orders from '../../components/details/Orders';
import Personal from '../../components/details/Personal';

const Details = ({ user }) => {
  const { first_name, avatar } = user;
  return (
    <div className={styles.container}>
      <Title text="User Details" customClassName={styles.title} />
      <div className={styles.avatar}>
        <Avatar alt={first_name} src={avatar} />
      </div>
      <div className={styles.personal}>
        <Personal user={user} />
      </div>
      <Orders />
      <Charts />
    </div>
  );
};

export default Details;
