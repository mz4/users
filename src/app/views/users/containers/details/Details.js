import { Title } from '../../../../library/index';
import Charts from '../../components/details/Charts';
import styles from './Details.module.scss';
import Orders from '../../components/details/Orders';

const Details = ({ avatar, personal }) => {
  return (
    <div className={styles.container}>
      <Title text="User Details" customClassName={styles.title} />
      <div className={styles.avatar}>{avatar}</div>
      <div className={styles.personal}>{personal}</div>
      <Orders />
      <Charts />
    </div>
  );
};

export default Details;
