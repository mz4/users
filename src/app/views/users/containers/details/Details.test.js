import renderer from 'react-test-renderer';
import Details from './Details';

const user = {
  id: 1,
  first_name: 'Mark',
  last_name: 'Guy',
  avatar: 'https://reqres.in/img/faces/1-image.jpg'
};

describe('Render component', () => {
  it('Create a snapshot', () => {
    const details = <Details user={user} />;
    const component = renderer.create(details);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
