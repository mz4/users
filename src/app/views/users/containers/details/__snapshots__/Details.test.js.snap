// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Create a snapshot 1`] = `
<div
  className="container"
>
  <div
    className="pageTitle title"
  >
    <span>
      User Details
    </span>
  </div>
  <div
    className="avatar"
  />
  <div
    className="personal"
  />
  <div
    className="box"
  >
    <div
      className="pageSubTitle"
    >
      <span>
        Total Orders Today
      </span>
    </div>
    <div
      className="orderValue"
    >
      12.450 $
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="pageSubTitle"
    >
      <span>
        Total Clients Today
      </span>
    </div>
    <div
      className="orderValue"
    >
      25
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="pageSubTitle"
    >
      <span>
        Revenue Day Ratio
      </span>
    </div>
    <div
      className="orderValue"
    >
      25%
    </div>
  </div>
  <div
    className="charts"
  >
    <div
      className="pageSubTitle"
    >
      <span>
        Chart Area
      </span>
    </div>
    <div
      className="recharts-responsive-container"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="charts"
  >
    <div
      className="pageSubTitle"
    >
      <span>
        Chart Billing
      </span>
    </div>
    <div
      className="recharts-responsive-container"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
